### PR TITLE
Move files to trash instead of deleting them

### PR DESCRIPTION
**Pull request by erikjv**

### CODE REVIEW

File: src/gui/folder.cpp [hunk 1]
```diff
@@ -1007,6 +1007,12 @@ void Folder::setDirtyNetworkLimits()
     _engine->setNetworkLimits(uploadLimit, downloadLimit);
 }
 
+void Folder::reloadSyncOptions()
+{
+    // Q: abort the sync and restart afterwards?
```
[the remark this answers is not in the document]
Comment (erikjv) [reply src/gui/folder.cpp]: We decided that we will not abort the sync.

File: src/gui/generalsettings.cpp [hunk 2]
```diff
@@ -80,6 +80,11 @@ GeneralSettings::GeneralSettings(QWidget *parent)
 
     _ui->crashreporterCheckBox->setVisible(Theme::instance()->withCrashReporter());
 
+    connect(_ui->moveToTrashCheckBox, &QCheckBox::toggled, this, [](bool checked) {
+        ConfigFile().setMoveToTrash(checked);
+        FolderMan::instance()->reloadSyncOptions();
+    });
```
Comment (TheOneRing) [inline src/gui/generalsettings.cpp:86]: Hmm no strong opinion but wouldn't a "options changed" signal be Qt'er?

Comment (erikjv) [reply src/gui/generalsettings.cpp:86]: This is about the same as done in `NetworkSettings::saveBWLimitSettings()` where a call is made to `FolderMan::instance()->setDirtyNetworkLimits();`

Comment (TheOneRing) [reply src/gui/generalsettings.cpp:86]: setDirtyNetworkLimits is 10 years old :D 

I'd prefer a `Q_EMIT syncOptionsChanged()` signal.
and a 
```
connect(foo, &GeneralSettings::syncOptionsChanged, _engine, []{ setSyncOptions(loadSyncOptions());});
````
or something like that?

**Review by TheOneRing — DISMISSED** [review]

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**